lib/chef_git/branched_data_bag.rb

```diff
@@ -0,0 +1,32 @@
+require 'chef/node'
+require 'mixlib/shellout'
+
+module ChefGit::BranchedDataBag
+  extend self
+
+  def bag_name(data_bag)
+    if chef_git_environment == 'master'
+      return data_bag
+    elsif data_bag_changed?(data_bag)
+      return "#{chef_git_environment}__#{data_bag}"
+    else
+      return data_bag
+    end
+  end
+
+  private
+
+  def data_bag_changed?(data_bag)
+    Dir.chdir(ChefGit::REPO_PATH) do
+      git_diff = Mixlib::ShellOut.new("git diff origin/master data_bags/#{data_bag}")
+      git_diff.run_command
+      unchanged = !git_diff.error? && git_diff.stdout.empty? && git_diff.stderr.empty?
+      return !unchanged
+    end
+  end
+
+  def chef_git_environment
+    node = Chef::Node.load(Chef::Config[:node_name])
```

Comment:
Does this load from the chef server, or does it apply `-E` from the command line as well?